src/lib/users/dialogs/inst_user_first.rb

```diff
@@ -425,10 +425,24 @@ def process_new_user_form
       @full_name = UI.QueryWidget(Id(:full_name), :Value)
 
       user.name = @username
+      user.home = Y2Users::Home.new("/home/#{user.name}")
```

Comment:
no escaping is needed?

Reply:
and why is this default here? I kind of expect that user propose this home as default value.

Reply:
I don't think so. According to *useradd* man page:

> Usernames must start with a lower case letter or an underscore, followed by lower case letters, digits, underscores, or dashes. They can end with a dollar sign. In regular expression terms: [a-z_][a-z0-9_-]*[$]?
>
> Usernames may only be up to 32 characters long.

The only way to use other characters for login is with *--badname* option, but we don't use such an option:

> --badname Allow names that do not conform to standards.

Reply:
> and why is this default here? I kind of expect that user propose this home as default value.

This is something we can add to `User`. For now, the user object is initialized without home. Maybe we should add some constructor methods:

# `#new`: with home and system `false.`
* `#create_system`: without home and system `true`.

Anyway, in this dialog we have to recreate the home, because in case of editing a user (when going back in the installation/firstboot), the home path should be adapted to the new user name (just in case the user name was modified).

Reply:
I guess we can also have a method in `User` to obtain the default home or something like that, in order to have the default generation (i.e., "/home/#{user.name}") only in one place.

Reply:
yeap, exactly my idea. We should have one place for default value for home. if it is spread over multiple clients, it is wrong from my POV

Reply:
Yes, completely agree. I would have a `User#default_home_path` to return the string and `User#default_home` to return the object.